logstash-core/lib/logstash/config/mixin.rb

```diff
@@ -126,6 +146,42 @@ def config_init(params)
     @config = params
   end # def config_init
 
+  # Replace all environment variable references in 'value' param by environment variable value and return updated value
+  # Process following patterns : $VAR, ${VAR}, ${VAR:defaultValue}
+  def replace_env_placeholders(value)
+    if (value.is_a?(String))
```

Comment:
If 'value' is not a string, the method will raise an error because it uses 'string' methods on 'value'.
Maybe it would be cleaner to make this test at line 117 ?